crates/cadenza-syntax/STATUS.md

```diff
@@ -0,0 +1,450 @@
+- Parenthesized expressions
+- Indentation-based blocks
+- Apply nodes for function application
+- Whitespace significance tracking
+- Generated binding power system
+
+⚠️ **Partially Complete:**
+- **Records**: Shorthand syntax works (`{ x, y }`), but field assignments (`{ x = 1 }`) fail due to parser marker propagation issue
+- **Error Recovery**: Basic error nodes exist but recovery is weak
+
+❌ **Not Implemented:**
+- Array indexing (`arr[0]`)
+- Quote/unquote operators (syntax not finalized)
+- Tuples vs grouping distinction
+- Prefix negation (`-x`) - intentionally removed due to ambiguity
+- Match expressions
+- If/else expressions
+- Function/closure syntax
+- Loops (while, for, loop)
+- Partial function application (`&foo arg`)
+- String interpolation
+
+## Detailed Status by Feature
+
+### 1. Operators ✅ MOSTLY COMPLETE
+
+**Status:** All operator categories implemented with proper binding powers.
+
+**Completed:**
+- Generated binding power system with enums
+- All infix operators (pipe, range, assignment, logical, bitwise, shift, arithmetic, exponentiation, field/path access)
+- Postfix operators (`?`, `|?`)
+- Prefix operators (`@`, `!`, `~`, `$`, `...`)
+
+**Known Issues:**
+- Prefix negation (`-x`) removed due to ambiguity with infix minus
+- Requires context-aware parsing to distinguish `-x` (negation) from `a - x` (subtraction)
+
+**References:** `PARSER_ISSUES.md` Issue 1
+
+---
+
+### 2. Quote/Unquote ❌ NOT IMPLEMENTED
+
+**Status:** Infrastructure missing, syntax not finalized.
+
+**Design Decision Needed:**
+- Current tokens: `'` (SingleQuote), `~` (Tilde)
+- Single quote conflicts with potential character literals
+- Alternative syntaxes to consider:
+  - Backtick for quote: `` `expr ``
+  - Dollar for unquote: `$expr`
+  - Keyword-like: `quote expr` / `unquote expr`
+
+**What's Needed:**
+1. Finalize syntax decision
+2. Add as prefix operators in parser
+3. Represent as `Apply(__quote__, [expr])` and `Apply(__unquote__, [expr])`
+4. Handle nested quotes and unquotes
+5. Support block quotes with indentation
+
+**References:** `PARSER_ISSUES.md` Issue 2
+
+---
+
+### 3. Array Literals ✅ COMPLETE
+
+**Status:** Fully implemented with comma-separated elements.
+
+**Working:**
+- Empty arrays: `[]`
+- Single/multiple elements: `[1]`, `[1, 2, 3]`
+- Nested arrays: `[[1, 2], [3, 4]]`
+- Trailing commas: `[1, 2,]`
+- Complex expressions: `[a + b, c * d]`
+
+**Representation:** `Apply(__list__, [elements...])`
+
+**Test Files:** `array-*.cdz` in test-data/
+
+---
+
+### 4. Array Indexing ❌ NOT IMPLEMENTED
+
+**Status:** No postfix bracket handling.
+
+**What's Needed:**
+1. Add `LBracket` as postfix operator in binding power system
+2. Parse expression inside brackets as index
+3. Represent as `Apply(__index__, [array, index])`
+4. Handle chained indexing: `arr[0][1]`
+5. Handle complex expressions: `arr[i + 1]`
+
+**Design Challenge:** Distinguish `[` as array literal prefix vs indexing postfix
```

Comment:
I think we can use whitespace for this. Basically if there's no space then it should be treated as an index. If there is a space then it's going to be a list.